contributors/design-proposals/thirdpartyresources.md

```diff
@@ -0,0 +1,194 @@
+# ThirdPartyResources - 2
+
+## Background
+There are numerous, fundamental issues with TPRs today that we’ve tried to keep
+track of [here](https://github.com/kubernetes/features/issues/95).  This design
+describes how to build a TPR implementation that resolves the known issues.
+
+
+## Goals
```

Comment:
Would do bullet list for goals and non-goals then give more data later

1. Ensure ThirdPartyResource authors can version their APIs and offer their users a safe migration path between versions (including alpha->beta for the TPR, but also including a discussion of how a TPR author could perform a migration for their end users)
2. Enable ThirdPartyResources to specify how they will appear in API discovery to be consistent with other resources and avoid naming confilcts
3. Move TPR into their own API group to allow the extensions group to be removed (link to removing extensions)
4. Support cluster scoped TPR resources
5. Identify other features required for TPR to become beta

Non-goals
 
1. Solve automatic conversion of TPR between versions or automatic migration of existing TPR